R/project.R

```diff
@@ -101,29 +101,12 @@ renv_project_type_impl <- function(path) {
 
 renv_project_remotes <- function(project, fields = NULL) {
 
-  # if this project has a DESCRIPTION file, use it to provide records
   descpath <- file.path(project, "DESCRIPTION")
-  if (file.exists(descpath))
-    return(renv_project_remotes_description(project, descpath, fields))
-
-  # otherwise, use the set of (non-base) packages used in the project
-  deps <- dependencies(
-    path     = project,
-    progress = FALSE,
-    errors   = "ignored",
-    dev      = TRUE
-  )
-
-  packages <- sort(unique(deps$Package))
-  ignored <- c("renv", renv_project_ignored_packages(project))
-  setdiff(packages, ignored)
-
-}
-
-renv_project_remotes_description <- function(project, descpath, fields = NULL) {
+  if (!file.exists(descpath))
```

Comment:
IIUC this is safe since the only time version-specific constraints could show up would be in the DESCRIPTION file; is that worth including as a comment? (as to why we don't just use `dependencies()`)

Reply:
Yeah, this function is now specifically about getting package constraints from the `DESCRIPTION`, rather than also finding package names from code.